### ISSUE TITLE
500 error at /signup/

### ISSUE THREAD

**Issue by bakert** [PennyDreadfulMTG/Penny-Dreadful-Tools#4374]
```
Failed to execute SELECT d.id, d.name AS original_name, d.created_date, d.updated_date, SUM(CASE WHEN dm.games > IFNULL(odm.games, 0) THEN 1 ELSE 0 END) AS wins, SUM(CASE WHEN dm.games < odm.games THEN 1 ELSE 0 END) AS losses, SUM(CASE WHEN dm.games = odm.games THEN 1 ELSE 0 END) AS draws, d.finish, d.archetype_id, d.url AS source_url, d.competition_id, c.name AS competition_name, c.end_date AS competition_end_date, ct.name AS competition_type_name, d.identifier, LOWER(IFNULL(IFNULL(IFNULL(p.name, p.mtgo_username), p.mtggoldfish_username), p.tappedout_username)) AS person, p.id AS person_id, p.banned, p.discord_id, d.created_date ASdate`,
d.decklist_hash,
d.retired,
s.name AS source_name,
IFNULL(a.name, '') AS archetype_name,
cache.normalized_name AS name,
cache.colors,
cache.colored_symbols,
cache.legal_formats
FROM
deck AS d
LEFT JOIN
person AS p ON d.person_id = p.id
LEFT JOIN
source AS s ON d.source_id = s.id
LEFT JOIN
archetype AS a ON d.archetype_id = a.id

    LEFT JOIN
        competition AS c ON d.competition_id = c.id
    LEFT JOIN
        competition_series AS cs ON cs.id = c.competition_series_id
    LEFT JOIN
        competition_type AS ct ON ct.id = cs.competition_type_id

    LEFT JOIN
        deck_cache AS cache ON d.id = cache.deck_id
    LEFT JOIN
        deck_match AS dm ON d.id = dm.deck_id
    LEFT JOIN
        deck_match AS odm ON odm.deck_id <> d.id AND dm.match_id = odm.match_id
    WHERE
        
    d.id IN (
        SELECT
            id
        FROM
            deck
        WHERE
            competition_id = (
    SELECT id FROM competition
    WHERE
        start_date < 1519953029.146068
    AND
        end_date > 1519953029.146068
    AND
        id IN (
    SELECT
        id
    FROM
        competition
    WHERE
        competition_series_id IN
            (
                SELECT
                    id
                FROM
                    competition_series
                WHERE
                    competition_type_id
                IN (
    SELECT
        id
    FROM
        competition_type
    WHERE
        name = 'League'
)
            )
    )
    )
    ) AND (
            SELECT
                COUNT(id)
            FROM
                deck_match AS dm
            WHERE
                dm.deck_id = d.id
        ) <= 4
    AND
        NOT d.retired
    AND
        (p.mtgo_username = 'goo\')

    GROUP BY
        d.id
    ORDER BY
        d.created_date DESC, d.finish IS NULL, d.finish
    
` with `[]` because of `(1064, "You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ''goo\\')\n    \n        GROUP BY\n            d.id\n        ORDER BY\n            d.cr' at line 103")`
Reported on decksite by logged_out
Request Method: POST
Path: /signup/?
Cookies: {'__cfduid': 'd5fad562e3b2fc07d544ecf4ab45307c51516748590', '_ga': 'GA1.2.395751054.1516748593', '_gid': 'GA1.2.170565212.1519952399', '_gat_gtag_UA_109131120_1': '1'}
Endpoint: add_signup
View Args: {}
Person: logged_out
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.186 Safari/537.36
Referrer: https://pennydreadfulmagic.com/signup/
Request Data: {'mtgo_username': 'goo\', 'name': '', 'decklist': ''}
DatabaseException
Stack Trace:
File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1997, in call
return self.wsgi_app(environ, start_response)
File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1985, in wsgi_app
response = self.handle_exception(e)
File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1982, in wsgi_app
response = self.full_dispatch_request()
File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1614, in full_dispatch_request
rv = self.handle_user_exception(e)
File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1517, in handle_user_exception
reraise(exc_type, exc_value, tb)
File "/home/discord/.local/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise
raise value
File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1612, in full_dispatch_request
rv = self.dispatch_request()
File "/home/discord/.local/lib/python3.6/site-packages/flask/app.py", line 1598, in dispatch_request
return self.view_functionsrule.endpoint
File "./decksite/cache.py", line 65, in decorated_function
response = make_response(f(*args, **kwargs))
File "./decksite/main.py", line 270, in add_signup
if form.validate():
File "./decksite/league.py", line 26, in validate
self.do_validation()
File "./decksite/league.py", line 41, in do_validation
elif active_decks_by(self.mtgo_username):
File "./decksite/league.py", line 172, in active_decks_by
return active_decks('p.mtgo_username = {mtgo_username}'.format(mtgo_username=sqlescape(mtgo_username, force_string=True)))
File "./decksite/league.py", line 168, in active_decks
decks = deck.load_decks(where)
File "./decksite/data/deck.py", line 147, in load_decks
rows = db().execute(sql)
File "./shared/database_mysql.py", line 52, in execute
raise DatabaseException('Failed to execute {sql} with {args} because of {e}'.format(sql=sql, args=args, e=e))
```

**Comment by jgabrielygalan** [comment]
Sanitize sql input? The user provided input ending in backslash, are we not using a method that sqlescapes user input?

**Comment by bakert** [comment]
shared.database.sqlescape is called on the input but it seemingly has a bug with backslashes.

**Comment by jgabrielygalan** [comment]
Fixed with #4832
